### DISCUSSION TITLE
Azure Active Directory authentication in .NET Aspire Distributed Application

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#114] [Announcements]
# Azure Active Directory authentication in .NET Aspire Distributed Application

https://vladislavantonyuk.github.io/articles/Azure-Active-Directory-authentication-in-.NET-Aspire-Distributed-Application/

**Comment by Conbag93** [comment]
Nice write up, but I couldn't get this working.

**Reply by Conbag93** [reply]
To be more specific...

When I try access a page which calls the API, I get the following error:

```
Microsoft.Identity.Web.TokenAcquisition: Information: [MsIdWeb] An error occured during token acquisition: No account or login hint was passed to the AcquireTokenSilent call. 

MSAL.NetCore.4.61.3.0.MsalUiRequiredException:
	ErrorCode: user_null
Microsoft.Identity.Client.MsalUiRequiredException: No account or login hint was passed to the AcquireTokenSilent call. 
```

At no point am I presented with the Azure AD B2C login screen.


P.S. On an unrelated note, you should allow users to edit their comments, if possible.

**Reply by VladislavAntonyuk** [reply]
You need to catch the exception and using MsalHandler handle it. Also make sure you provide scopes. It is required.

You can edit comments on GitHub, I use giscus for comments and am limited by its features.
